### PR TITLE
framework-tool: init at unstable-2023-11-14

### PR DESCRIPTION
**Pull request by io12**
## Description of changes

Add `framework_tool` from https://github.com/FrameworkComputer/framework-system/.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/fr/framework-tool/package.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchFromGitHub {
+    owner = "FrameworkComputer";
+    repo = "framework-system";
+    rev = "491a587342a5d79366a25d803b7065169314279c";
+    hash = "sha256-qDtW4DvY19enCfkOBRaako9ngAkmSreoNWlL4QE2FAk=";
+  };
+
+  cargoLock = {
+    lockFile = "${src}/Cargo.lock";
```
Comment (NickCao) [inline pkgs/by-name/fr/framework-tool/package.nix:18]: This causes IFD (import from derivation), which is not allowed in nixpkgs, you may copy the lockfile instead.

Comment (io12) [reply pkgs/by-name/fr/framework-tool/package.nix]: Fixed, thanks!

**Review by NickCao — APPROVED** [review]
diff LGTM, cannot test though (still waiting for my batch 8 to arrive!)

File: pkgs/by-name/fr/framework-tool/package.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ udev ];
+
+  meta = with lib; {
+    description = "Swiss army knife for Framework laptops";
+    homepage = "https://github.com/FrameworkComputer/framework-system";
+    license = licenses.bsd3;
+    platforms = platforms.linux;
+    mainProgram = "framework_tool";
```
Comment (NickCao) [inline pkgs/by-name/fr/framework-tool/package.nix:34]: Also: would you like to add your self as a maintainer of this package?

Comment (io12) [reply pkgs/by-name/fr/framework-tool/package.nix:34]: No thanks. But I think it's pretty simple and doesn't need that much maintenance.

Comment (NickCao) [reply pkgs/by-name/fr/framework-tool/package.nix:34]: Well actually there must be a maintainer for new packages: https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes, feel free to add me to it.

`maintainers = with maintainers; [ nickcao ];`

File: pkgs/by-name/fr/framework-tool/package.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ udev ];
+
+  meta = with lib; {
+    description = "Swiss army knife for Framework laptops";
+    homepage = "https://github.com/FrameworkComputer/framework-system";
+    license = licenses.bsd3;
+    platforms = platforms.linux;
```
Comment (NickCao) [inline pkgs/by-name/fr/framework-tool/package.nix:33]: It does not build for aarch64-linux, so maybe `platforms = [ "x86_64-linux" ];`

**Comment by NickCao** [comment]
Don't forget to squash the commits after everything.

**Comment by io12** [comment]
> Don't forget to squash the commits after everything.

Done

**Review by NickCao — APPROVED** [review]

**Pull request merged**